src/audio/module_adapter/module_adapter.c

```diff
@@ -35,7 +35,21 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	int ret;
 	struct comp_dev *dev;
 	struct processing_module *mod;
-	struct ipc_config_process *ipc_module_adapter = spec;
+	unsigned char *data;
+	uint32_t size;
+
+	switch (config->type) {
+	case SOF_COMP_MODULE_ADAPTER:
+	{
+		struct ipc_config_process *ipc_module_adapter = spec;
+
+		size = ipc_module_adapter->size;
+		data = ipc_module_adapter->data;
+		break;
+	}
+	default:
+		return NULL;
```

Comment:
@ranj063 having a second thought about this I think we shouldn't use `config->type` but UUID as we agreed with @cujomalainey at some point.

We want to avoid adding new component types in the future.

Reply:
Personally I want to delete that enum, i think its just confusing for newcomers.

Reply:
Pointer *spec points directly to IPC mailbox address. In this code You assume, it maps to following structure:
struct ipc_config_process {
	uint32_t size;	/**< size of bespoke data section in bytes */
	uint32_t type;	/**< sof_ipc_process_type */
	unsigned char *data;
};
I think this is not true for IPC4 protocol, where initial data are passed through IPC4 header extended, and payload starts with beginning of the configuration. The size variable will be wrong here.

Reply:
@jxstelter right the module adapter needs to be adapter for IPC4. My next target is to convert the copier. I will handle this with that change